### PR TITLE
PM-2165 Migrate delete account dialog

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate the Delete account component to use the component library.

## Code changes

- **delete-account.component.html:** Updated modal to its equivalent dialog components of library and buttons also updated.
- **delete-account.component.ts:** Added a static method for dialog open from account component. Updated submit method.
- **account.component.html:** Updated delete account button and added bit action to it.
- **account.component.ts:** Delete account dialog open logic implemented.

## Screenshots
**Delete Account Dialog**
Before:
image
After:
image

**Screen after account Delete**
Before:
image
After:
image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/web/src/app/auth/settings/account/delete-account.component.ts [hunk 1]
```diff
@@ -23,21 +22,21 @@ export class DeleteAccountComponent {
...
+    this.platformUtilsService.showToast(
+      "success",
+      this.i18nService.t("accountDeleted"),
+      this.i18nService.t("accountDeletedDesc"),
+    );
+  };
+
+  static openDeleteAccountDialog(dialogService: DialogService) {
+    return dialogService.open(DeleteAccountComponent);
   }
```
Comment (willmartian) [inline apps/web/src/app/auth/settings/account/delete-account.component.ts:41]: Could you rename this static method to just `open`?

File: apps/web/src/app/auth/settings/account/delete-account.component.ts [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
 import { Verification } from "@bitwarden/common/auth/types/verification";
 import { I18nService } from "@bitwarden/common/platform/abstractions/i18n.service";
-import { LogService } from "@bitwarden/common/platform/abstractions/log.service";
 import { PlatformUtilsService } from "@bitwarden/common/platform/abstractions/platform-utils.service";
+import { DialogService } from "@bitwarden/components";
 
 @Component({
   selector: "app-delete-account",
   templateUrl: "delete-account.component.html",
 })
```
Comment (willmartian) [inline apps/web/src/app/auth/settings/account/delete-account.component.ts:14]: 1. The selector isn't needed and should be removed
2. Could you update the component name to reference that it is a dialog? `DeleteAccountDialogComponent`

**Review by willmartian — CHANGES_REQUESTED** [review]
Looks largely good to me, just a couple change requests:

File: apps/web/src/app/auth/settings/account/delete-account-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<form [formGroup]="deleteForm" [bitSubmit]="submit">
+  <bit-dialog dialogSize="default" [title]="'deleteAccount' | i18n">
+    <ng-container bitDialogContent>
+      <p bitTypography="body1">{{ "deleteAccountDesc" | i18n }}</p>
+      <app-callout type="warning">{{ "deleteAccountWarning" | i18n }}</app-callout>
+      <app-user-verification
+        formControlName="verification"
+        name="verification"
+      ></app-user-verification>
```
Comment (coroiu) [inline apps/web/src/app/auth/settings/account/delete-account-dialog.component.html:9]: ❌ issue: `app-user-verification` is deprecated. Use `app-user-verification-form-input` instead.

❌ issue: both of these components currently need the parent component to keep track of `invalidSecret` to be able to properly display an error if the user has entered a wrong secret. 

Here is an example where the new component is used: https://github.com/bitwarden/clients/blob/a4b15bbcf081dca13a057c444660af12cdbbf0b7/libs/auth/src/angular/user-verification/user-verification-dialog.component.html#L41-L47

File: apps/web/src/app/auth/settings/account/delete-account-dialog.component.ts [hunk 4]
```diff
@@ -23,21 +21,21 @@ export class DeleteAccountComponent {
...
+  submit = async () => {
+    const verification = this.deleteForm.get("verification").value;
+    await this.accountApiService.deleteAccount(verification);
+    this.dialogRef.close();
+    this.platformUtilsService.showToast(
+      "success",
+      this.i18nService.t("accountDeleted"),
+      this.i18nService.t("accountDeletedDesc"),
+    );
+  };
```
Comment (coroiu) [inline apps/web/src/app/auth/settings/account/delete-account-dialog.component.ts:36]: 🎨 suggestion: should we maybe disable dialog-closing while this request is in progress?

**Review by coroiu — CHANGES_REQUESTED** [review]

File: apps/web/src/app/auth/settings/account/delete-account-dialog.component.html (hunk 3 above)
Comment (KiruthigaManivannan) [reply apps/web/src/app/auth/settings/account/delete-account-dialog.component.html:9]: @coroiu  Earlier it was mentioned in JIRA that we need to use new User verification dialog component . And I had few queries in using that.  
https://bitwarden.atlassian.net/browse/PM-2165. Todd has given some comments in that. And after going through the analysis document he edited the JIRA description also. 
And also when we had some issues in using new app-user-verification-form-input  in other two factor verify dialog stuff he mentioned the below message in slack.

image

But for this component can we check once again. ?? Whether to use new app-user-verification -form-input or old one.

Comment (coroiu) [reply apps/web/src/app/auth/settings/account/delete-account-dialog.component.html:9]: Oh interesting, I will check and get back to you!

Comment (coroiu) [reply apps/web/src/app/auth/settings/account/delete-account-dialog.component.html:9]: Oh interesting! @trmartin4 @jlf0dev @JaredSnider-Bitwarden can you answer this?

Comment (trmartin4) [reply apps/web/src/app/auth/settings/account/delete-account-dialog.component.html]: @coroiu I believe we covered this in Slack, but we are actually good with using the new component, and it has been migrated on this PR.  Thank you for pushing for it!

**Review by coroiu — COMMENTED** [review]
> ❌ issue: both of these components currently need the parent component to keep track of invalidSecret to be able to properly display an error if the user has entered a wrong secret.
> - https://github.com/bitwarden/clients/pull/8503#discussion_r1557540540

Missing `invalidSecret` handling

**Example:**
Recording 2024-05-13 at 16 07 31

**This PR:**
Recording 2024-05-13 at 16 09 47

**Comment by KiruthigaManivannan** [comment]
https://github.com/bitwarden/clients/pull/8503#pullrequestreview-2052821744  will address this soon @coroiu

**Comment by KiruthigaManivannan** [comment]
@trmartin4 I have added invalidSecret to form input as per your advise provided in slack. Is this code change fine

**Review by coroiu — DISMISSED** [review]
Looking good!

**Review by jlf0dev — APPROVED** [review]

**Review by coroiu — APPROVED** [review]

**Pull request merged**